session/git/worktree_ops.go

```diff
@@ -57,6 +63,12 @@ func (g *GitWorktree) SetupNewWorktree() error {
 		return fmt.Errorf("failed to create worktrees directory: %w", err)
 	}
 
+	// Ensure the parent directory of the worktree path exists
+	worktreeParent := filepath.Dir(g.worktreePath)
```

Comment:
Can you please explain why this is needed?

Reply:
Well `git worktree add` requires a directory to create worktree. Since the `SetupNewWorktree` and `SetupFromExistingBranch` are separately called in the `Setup` we need do do this in both of them.

So this checks for the directory and creates it if it does not exist to then be able to perform the `git worktree add`.